### PR TITLE
feat(snapshots): snapshot data is now retrieved and processed

### LINKED ISSUE #21: vSphere Snapshot information
As reported #20  it would a nice addition for users to have access to Snapshot information.

Data should be map to a new entity and data can be fetched making use of the same client and API:
 - [VMware object](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/790263bc-bd30-48f1-af12-ed36055d718b/e5f17bfc-ecba-40bf-a04f-376bbb11e811/vim.vm.Snapshot.html) representing snapshot

### PR DESCRIPTION
**Pull request by paologallinaharbur**
This PR aims to add the support to fetch and process snapshot 
 - Added support for snapshots
 - Added tests
 - Added flag to enable/disable feature
```EnableVsphereSnapshots bool   default:"true" help:"If set to true integration will collect, process and send as well data regarding vm Snapshots"```

fix #21

### CODE REVIEW

File: internal/process/process.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ const (
 	entityTypeResourcePool = "ResourcePool"
 	entityTypeVm           = "Vm"
 	entityTypeDatastore    = "Datastore"
+	entityTypeSnapshotVm   = "SnapshotVm"
```
Comment (gsanchezgavier) [inline internal/process/process.go:22]: this confuses me a bit since there is no entity for snapshot

File: internal/process/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		traverseSnapshotList(e, config, s, treeInfo, info)
+	}
+
+}
+
+func createMetricsCurrentSnapshot(treeInfo string, tree types.VirtualMachineSnapshotTree, config *load.Config, ms *metric.Set, info map[types.ManagedObjectReference]*infoSnapshot) {
+	checkError(config, ms.SetMetric("snapshotTreeInfo", treeInfo, metric.ATTRIBUTE))
+	checkError(config, ms.SetMetric("name", tree.Name, metric.ATTRIBUTE))
+	checkError(config, ms.SetMetric("creationTime", tree.CreateTime.String(), metric.ATTRIBUTE))
+	checkError(config, ms.SetMetric("powerState", string(tree.State), metric.ATTRIBUTE))
```
Comment (gsanchezgavier) [inline internal/process/snapshot.go:85]: should't we add the powerState(in an aggregated way) in the vm event in order to alert on vm that are running from snapshots?

Comment (paologallinaharbur) [reply internal/process/snapshot.go:85]: the power state attributes indicates the status of the machine when the snapshot was taken, not the power status of the snapshot

File: internal/process/process.go (hunk 1 above)
Comment (paologallinaharbur) [reply internal/process/process.go:22]: I will change the name!

**Pull request merged**